### PR TITLE
:tada: more flexible prefetch, supported array or callback

### PR DESCRIPTION
**Pull request by howel52**
##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

- resolve https://github.com/umijs/qiankun/issues/293
- supported array or callback

### CODE REVIEW

File: docs/zh/api/README.md [hunk 1]
```diff
@@ -80,9 +80,14 @@
...
-    - prefetch - `boolean | 'all'` - 可选，是否开启预加载，默认为 `true`。
+    - prefetch - `boolean | 'all' | string[] | (( apps: RegistrableApp[] ) => { mainAppStartingAppsName: string[]; firstMountedAppsName: string[] })`
+    - 可选，是否开启预加载，默认为 `true`。
 
-      配置为 `true` 则会在第一个子应用 mount 完成后开始预加载其他子应用的静态资源。配置为 `'all'` 则主应用 `start` 后即开始预加载所有子应用静态资源。
+      配置为 `true` 则会在第一个子应用 mount 完成后开始预加载其他子应用的静态资源，配置为 `'all'` 则主应用 `start` 后即开始预加载所有子应用静态资源。
+
+      配置为 `string[]` 则会在第一个子应用 mount 完成侯哦开始加载数组内的子应用的资源
+
+      配置为 `function`  则可完全自定义应用的加载实际 (首个子应用 mounted，或主应用 starting)
```
Comment (kuitos) [inline docs/zh/api/README.md]: > 则可完全自定义应用的加载实际

typo?

File: src/register.ts [hunk 2]
```diff
@@ -194,21 +194,34 @@ export function registerMicroApps<T extends object = {}>(
   });
 }
 
+async function declarePrefetch(prefetch: Prefetch, importEntryOpts: ImportEntryOpts) {
```
Comment (kuitos) [inline src/register.ts]: declarePrefetch -> doPrefetch?

声明应该是表明不带副作用的

File: docs/zh/api/README.md [hunk 3]
```diff
@@ -80,9 +80,14 @@
...
   - `Options`
 
-    - prefetch - `boolean | 'all'` - 可选，是否开启预加载，默认为 `true`。
+    - prefetch - `boolean | 'all' | string[] | (( apps: RegistrableApp[] ) => { mainAppStartingAppsName: string[]; firstMountedAppsName: string[] })`
+    - 可选，是否开启预加载，默认为 `true`。
 
-      配置为 `true` 则会在第一个子应用 mount 完成后开始预加载其他子应用的静态资源。配置为 `'all'` 则主应用 `start` 后即开始预加载所有子应用静态资源。
+      配置为 `true` 则会在第一个子应用 mount 完成后开始预加载其他子应用的静态资源，配置为 `'all'` 则主应用 `start` 后即开始预加载所有子应用静态资源。
+
+      配置为 `string[]` 则会在第一个子应用 mount 完成侯哦开始加载数组内的子应用的资源
```
Comment (kuitos) [inline docs/zh/api/README.md]: > 完成侯哦开始加载数组
又发现一个 typo 👻

另外英文文档也补充一下？

File: src/interfaces.ts [hunk 4]
```diff
@@ -28,8 +28,14 @@ export type RegistrableApp<T extends object = {}> = {
   props?: T; // props pass through to app
 };
 
+export type Prefetch =
+  | boolean
+  | 'all'
+  | string[]
+  | ((apps: RegistrableApp[]) => { mainAppStartingAppsName: string[]; firstMountedAppsName: string[] });
```
Comment (kuitos) [inline src/interfaces.ts]: 这个最外层的括号是 prettier 加的吗？不是的话去掉吧

File: docs/zh/api/README.md (hunk 3 above)
Comment (howel52) [reply docs/zh/api/README.md]: 👌

**Pull request merged**